Fix closeText in Modal over screen and improve MediaModal hook

isacoder opened the pull request:
### Requirements
closes Bug #142 The video currently there is no limit to he height in modal probably we should limit it
closes Feature #155 - useMediaModal hook

### Implementation

- Updated MediaBox component to limit height and width space for modal but it's retro-compatible because this is only added if prop hasModalLimits is true
- Updated modal to have isModalOpen value available since it was mention that it would be helpful
- created useMediaModal hook to manage modal and MediaBox together with some strict values like width: auto, no padding and should always be close able.
- onError, onLoad props features for MediaBox
- Zoom in TableThumbnail only available if there is preview

currently this feature can be reviewed in the demo link.
https://future-standard.github.io/scorer-ui-kit/#/table

### Screenshoot

<img width="1164" alt="Screen Shot 2021-08-06 at 11 41 04" src="https://user-images.githubusercontent.com/10409078/128447905-049378c7-de97-41ad-b502-392575ec2701.png">

Video of interaction
Basically TableThumbnail component verifies preview link works if it doesn't zoom and click is not available. 
Video and image are now responsive width and hight.

https://drive.google.com/file/d/1Qe31NbonhWJB4oqB4rw4P7wlvbbxUwVN/view?usp=sharing

atomworks reviewed (changes requested):
There's a couple of issues I've noticed here so wanted to address those before going through the code.

- The disabled hover behaviour has some impact on usability. It's definitely good to disable it when there is no content at all (such as the final item in the example) but when there is thumbnail but no large version it is better to keep the zoom on hover which still has value. For implying a modal is available it's better to use `cursor` properties to do this.

- Seems like aspect ratio gets blown out and stretch on images when I checked it on linked Examples. I wanted to check the video here too but there didn't seem to be anything here.

![image](https://user-images.githubusercontent.com/808289/128969977-3685d6de-bbfb-4b6d-91dc-fffff523ec0e.png)

isacoder commented:
Welp.. I couldn't replicate but since we review it together I have updated in an attempt to try something in your setup..
basically max-width and max-height 100% base.

https://future-standard.github.io/scorer-ui-kit/#/table

Since we talked about it I have updated the modal in storybook and the Tables to have modals too.

https://future-standard.github.io/scorer-ui-kit/storybook/?path=/story/alerts-modals--media-modal

Previous version can be review in this Story by removing the prop ' Has Modal Prop'
You will need to retrigger the modal it since story book does not refresh this.

<img width="1079" alt="Screen Shot 2021-08-12 at 19 03 53" src="https://user-images.githubusercontent.com/10409078/129178677-c3457cce-430c-4469-beab-48c8ad929852.png">

atomworks reviewed (approved):
👍
